### PR TITLE
[Enhancement] wandb define_metric

### PR DESCRIPTION
**Pull request by okotaku**
## Motivation

Log best metric/loss with wandb.

## Use cases (Optional)

```
log_config = dict(
    interval=50,
    ignore_last=False,
    hooks=[
        dict(type='TextLoggerHook'),
        dict(type='WandbLoggerHook',
             define_metric_cfg=dict(bbox_mAP='max'),
             init_kwargs=dict(
                 project='example',
                 name='example001'
             ))
    ])
```

Example result

スクリーンショット 2022-08-30 14 51 53

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/runner/hooks/logger/wandb.py [hunk 1]
```diff
@@ -43,6 +43,10 @@ class WandbLoggerHook(LoggerHook):
             ``out_suffix`` will be uploaded to wandb.
             Default: ('.log.json', '.log', '.py').
             `New in version 1.4.3.`
+        define_metric_cfg (dict):
+            A dict of metrics and summary for wandb.define_metric.
+            Example. dict(bbox_mAP='max')
+            Default: None
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/wandb.py:49]:
```suggestion
        define_metric_cfg (dict, optional): A dict of metrics and summaries for
            wandb.define_metric. The key is metric and the value is summary.
            For example, if setting ``define_metric_cfg={'coco/bbox_mAP': 'max'}``,
            the maximum value of``coco/bbox_mAP`` will be logged on wandb UI.
            See `wandb docs <https://docs.wandb.ai/ref/python/run#define_metric>`_
            for details.
            Default: None.
            `New in version 1.6.2.`
```

**Comment by zhouzaida** [comment]
Hi @okotaku , thanks for your contribution. We need to add unit tests for this new parameter in https://github.com/open-mmlab/mmcv/blob/83bedf5a2b731dc1668b2100a860c16c01863bee/tests/test_runner/test_hooks.py#L1606.

File: mmcv/runner/hooks/logger/wandb.py [hunk 2]
```diff
@@ -83,6 +94,10 @@ def before_run(self, runner) -> None:
             self.wandb.init(**self.init_kwargs)  # type: ignore
         else:
             self.wandb.init()  # type: ignore
+        if self.define_metric_cfg is not None:
+            for metric, summary in self.define_metric_cfg.items():
```
Comment (ayulockin) [inline mmcv/runner/hooks/logger/wandb.py:98]: Thanks @okotaku for adding this feature to this base wandb logger. :) A couple of comments.
 
- We should ensure that the summary string passed as the value of `define_metric_cfg` dict should be in `["min", "max", "mean" ,"best", "last", "none"]`. If this is not the case, the user should get a warning, but it should not error out.

- Is there a way to ensure that the `metric` is part of the metrics that will be logged to W&B. I think you can access the metrics by doing this: `tags = self.get_loggable_tags(runner)`. Even if the `metric` is spelled wrong or wrongly provided, `wandb.define_metric` will not throw error but a check will be useful for users imo.

Comment (okotaku) [reply mmcv/runner/hooks/logger/wandb.py:102]: I think 2nd one is difficult.
The before run 'tags', train 'tags', val 'tags' are different from each other.

File: mmcv/runner/hooks/logger/wandb.py [hunk 3]
```diff
@@ -43,6 +44,17 @@ class WandbLoggerHook(LoggerHook):
             ``out_suffix`` will be uploaded to wandb.
             Default: ('.log.json', '.log', '.py').
             `New in version 1.4.3.`
+        define_metric_cfg (dict, optional): A dict of metrics and summaries for
+            wandb.define_metric. The key is metric and the value is summary.
+            The summary should be in ["min", "max", "mean" ,"best", "last",
+             "none"].
+            For example, if setting
+            ``define_metric_cfg={'coco/bbox_mAP': 'max'}``, the maximum value
+            of``coco/bbox_mAP`` will be logged on wandb UI. See
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/wandb.py:53]:
```suggestion
            of ``coco/bbox_mAP`` will be logged on wandb UI. See
```

File: mmcv/runner/hooks/logger/wandb.py [hunk 4]
```diff
@@ -43,6 +44,17 @@ class WandbLoggerHook(LoggerHook):
...
+            `wandb docs <https://docs.wandb.ai/ref/python/run#define_metric>`_
+            for details.
+            Default: None.
+            `New in version 1.6.2.`
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/wandb.py:57]:
```suggestion
            Defaults to None.
            `New in version 1.6.3.`
```

**Comment by zhouzaida** [comment]
Hi @okotaku , thanks for your contribution. This PR can be merged after resolving above comments.

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**